src/ValueParsers/DecimalParser.php

```diff
@@ -83,14 +128,7 @@ protected function stringParse( $value ) {
 		$value = $this->unlocalizer->unlocalizeNumber( $value );
 
 		//handle scientific notation
-		if ( preg_match( '/^(.*)([eE]|x10\^)([-+]?[,\d]+)$/', $value, $matches ) ) {
-			$exponent = $this->normalizeDecimal( $matches[3] );
-			$exponent = intval( $exponent );
-
-			$value = $matches[1];
-		} else {
-			$exponent = 0;
-		}
+		list( $value, $exponent ) = $this->splitDecimalExponent( $value );
 
 		$value = $this->normalizeDecimal( $value );
```

Comment:
Could do that, I suppose. Will have to double-check.
Note that the function is public so it can be used by QuantityValue.